### PR TITLE
Fix mimirtool error logging

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>

#### What this PR does

When provided with `--log.level=debug` and a request fails mimirtool should print out some part of the response for easier debugging. Instead it uses a scanner which only reads the first token from the response.

The first token is by default the first line. Which in some cases can be fairly useless.

This PR 
* changes the logged reported content to be the first 1024 bytes regardless of new lines.
* removes duplication of the error content (logging both `errMsg` and `msg`)
* increases the number of bytes to read from the body from 512 to 1024 for good measure.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- NA Tests updated
- NA Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,9 @@
 ### Jsonnet
 
 ### Mimirtool
+
 * [ENHANCEMENT] Added `mimirtool rules delete-namespace` command to delete all of the rule groups in a namespace including the namespace itself. #3136
+* [BUGFIX] `--log.level=debug` not correctly prints the response from the remote endpoint when a request fails. #3180
```
Comment (aknuds1) [inline CHANGELOG.md:18]:
```suggestion
* [BUGFIX] `--log.level=debug` correctly prints the response from the remote endpoint when a request fails. #3180
```

**Review by aknuds1 — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (dimitarvdimitrov) [reply CHANGELOG.md:18]: it should've been `now` 🤦

Comment (dimitarvdimitrov) [reply CHANGELOG.md:18]:
```suggestion
* [BUGFIX] `--log.level=debug` now correctly prints the response from the remote endpoint when a request fails. #3180
```

File: pkg/mimirtool/client/client.go [hunk 2]
```diff
@@ -174,36 +173,34 @@ func checkResponse(r *http.Response) error {
...
 	}
 
+	msg := string(bodyHead)
+	var errMsg string
 	if msg == "" {
-		errMsg = fmt.Sprintf("server returned HTTP status %s", r.Status)
+		errMsg = fmt.Sprintf("server returned HTTP status: %s", r.Status)
 	} else {
-		errMsg = fmt.Sprintf("server returned HTTP status %s: %s", r.Status, msg)
+		errMsg = fmt.Sprintf("server returned HTTP status: %s, body: %s", r.Status, msg)
```
Comment (bboreham) [inline pkg/mimirtool/client/client.go:186]: Maybe this should be `%q`, so any funny characters are visible?
I can see arguments on both sides whether newline or `\n` is easier for the reader, but a `\t` would definitely be clearer than some whitespace.

File: pkg/mimirtool/client/client.go [hunk 3]
```diff
@@ -174,36 +173,34 @@ func checkResponse(r *http.Response) error {
...
 		log.WithFields(log.Fields{
 			"status": r.Status,
-			"msg":    msg,
 		}).Debugln(errMsg)
 		return errConflict
 	}
 
 	log.WithFields(log.Fields{
 		"status": r.Status,
-		"msg":    msg,
```
Comment (bboreham) [inline pkg/mimirtool/client/client.go:206]: Why did this change?  I don't see it mentioned in the commit description.

Comment (dimitarvdimitrov) [reply pkg/mimirtool/client/client.go:206]: it doubles the error message. `errMsg` already contains `msg`

https://github.com/grafana/mimir/blob/465cbffd1a8b11841c003c86d03ac548e9cd847f/pkg/mimirtool/client/client.go#L180-L194

Comment (aknuds1) [reply pkg/mimirtool/client/client.go:206]: I noticed the same as @dimitarvdimitrov is saying, figured that's why `msg` gets removed.

Comment (dimitarvdimitrov) [reply pkg/mimirtool/client/client.go:206]: actually I went ahead and did some more changes. even with the removed `msg` (and with `%q`), there was still duplication

```
time="2022-10-11T07:28:28Z" level=error msg="server returned HTTP status: 400 Bad Request, body: \"block upload is disabled\\n\"" status="400 Bad Request"
```

in da755e9c073951a7eef0a40f6b86e13c021d1046 I changed this to 

```
time="2022-10-11T07:45:02Z" level=error msg=response body="block upload is disabled\n" status="400 Bad Request"
```

and an example flow is for a failed blocks upload

```
time="2022-10-11T07:45:02Z" level=info msg=Backfilling blocks=/shared/01GF301Z9T3CX0YK0VZ5B18MWJ user=anonymous
time="2022-10-11T07:45:02Z" level=info msg="making request to start block upload" block=01GF301Z9T3CX0YK0VZ5B18MWJ file=meta.json path=/shared/01GF301Z9T3CX0YK0VZ5B18MWJ
time="2022-10-11T07:45:02Z" level=error msg=response body="block upload is disabled\n" status="400 Bad Request"
time="2022-10-11T07:45:02Z" level=error msg="failed uploading block" error="request to start block upload failed: POST request to http://compactor:8080/api/v1/upload/block/01GF301Z9T3CX0YK0VZ5B18MWJ/start failed: server returned HTTP status: 400 Bad Request, body: \"block upload is disabled\\n\"" path=/shared/01GF301Z9T3CX0YK0VZ5B18MWJ
time="2022-10-11T07:45:02Z" level=info msg="finished uploading blocks" already_exists=0 failed=1 succeeded=0
mimirtool: error: blocks failed to upload 1 block(s), try --help
```

**Comment by dimitarvdimitrov** [comment]
can you please take another look @bboreham?

**Review by pracucci — APPROVED** [review]
This looks very useful to me. LGTM!

**Pull request merged**